### PR TITLE
fix(PE-3574): updated transaction complete links to route to correct pages, …

### PR DESCRIPTION
**Pull request by atticusofsparta**
…updated util to correctly copy steps object, updated create pdnt to use transaction workflow

### CODE REVIEW

File: src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx [hunk 1]
```diff
@@ -80,30 +78,40 @@ function TransactionWorkflow({
...
                   type: 'setDeployedTransactionId',
-                  payload: originalTxId,
+                  payload: new ArweaveTransactionID(originalTxId.toString()),
                 });
               }
-              newSteps['2'].status = 'success';
-              newSteps['3'].status = 'success';
-              setSteps(newSteps);
+              steps['2'].status = 'success';
+              steps['3'].status = 'success';
```
Comment (dtfiedler) [inline src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx:114]: nit: put this in a function and call it from the switch statement. it's hard to understand what's going on in here with it nested

File: src/utils/transactionUtils/transactionUtils.ts [hunk 2]
```diff
@@ -174,7 +195,7 @@ export const getWorkflowStepsForInteraction = (
     status: string;
   };
 } => {
-  return WorkflowStepsForInteractions[interaction];
+  return _.cloneDeep(WorkflowStepsForInteractions[interaction]);
```
Comment (dtfiedler) [inline src/utils/transactionUtils/transactionUtils.ts:198]: `structuredClone()` is a built-in function for deep-copying. lodash is a large package, so avoid it if we can

https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

File: src/utils/transactionUtils/transactionUtils.ts [hunk 3]
```diff
@@ -380,9 +390,9 @@ export function getInteractionTypeFromField(field: string) {
 
 export function mapTransactionDataKeyToPayload(
   interactionType: ValidInteractionType,
-  data: string | number | Array<string | number>,
+  data: string | number | Array<string | number | PDNTContractJSON>,
```
Comment (dtfiedler) [inline src/utils/transactionUtils/transactionUtils.ts:393]: i think we should just change this to an object vs. an array, but can be done in another PR.

File: src/utils/transactionUtils/transactionUtils.ts [hunk 4]
```diff
@@ -405,3 +415,28 @@ export function mapTransactionDataKeyToPayload(
...
+  if (
+    interactionType === INTERACTION_TYPES.CREATE &&
+    isObjectOfTransactionPayloadType<CreatePDNTPayload>(
+      transactionData,
+      TRANSACTION_DATA_KEYS[INTERACTION_TYPES.CREATE].keys,
+    )
+  ) {
+    return transactionData.deployedTransactionId?.toString() ?? '';
+  }
+  return transactionData.assetId.toString() ?? '';
```
Comment (dtfiedler) [inline src/utils/transactionUtils/transactionUtils.ts:441]: i don't think `assetId` can be null, so no null-coalesce needed here

**Review by dtfiedler — APPROVED** [review]
functionally looks good - loving seeing `/create` fold into the `/transaction` workflow. some small nits to cleanup.

File: src/utils/transactionUtils/transactionUtils.ts (hunk 3 above)
Comment (atticusofsparta) [reply src/utils/transactionUtils/transactionUtils.ts:393]: you may be right.... i was originally using this function for url params were all the data was stored in the url under the `data` search params, and was sticking with that logic here.

File: src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx (hunk 1 above)
Comment (atticusofsparta) [reply src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx:114]: added

File: src/utils/transactionUtils/transactionUtils.ts (hunk 2 above)
Comment (atticusofsparta) [reply src/utils/transactionUtils/transactionUtils.ts:198]: added

File: src/utils/transactionUtils/transactionUtils.ts (hunk 4 above)
Comment (atticusofsparta) [reply src/utils/transactionUtils/transactionUtils.ts:441]: added

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**